src/core/version_neg.c

```diff
@@ -0,0 +1,384 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    This file defines the logic for version negotiation.
+
+--*/
+
+#include "precomp.h"
+#ifdef QUIC_CLOG
+#include "version_neg.c.clog.h"
+#endif
+
+typedef struct QUIC_COMPATIBLE_VERSION_MAP {
+    const uint32_t OriginalVersion;
+    const uint32_t CompatibleVersion;
+} QUIC_COMPATIBLE_VERSION_MAP;
+
+const QUIC_COMPATIBLE_VERSION_MAP CompatibleVersionsMap[] = {
+    {QUIC_VERSION_DRAFT_29, QUIC_VERSION_DRAFT_29},
+    {QUIC_VERSION_MS_1, QUIC_VERSION_MS_1},
```

Comment:
We should just know versions are always compatible with themselves. No need to increase the size of this array for that.

Reply:
It makes the implementation of the loops cleaner by having the array like this.

Reply:
Perhaps, but I'd really prefer not having this constant. It shouldn't be _that_ hard to fix the loops to accept themselves as compatibile.